aks-auto-shutdown-permissions.tf

```diff
@@ -0,0 +1,7 @@
+resource "azurerm_role_assignment" "aks_auto_shutdown" {
+  count = var.aks_auto_shutdown_principal_id == "" ? 0 : 1
+
+  principal_id         = var.aks_auto_shutdown_principal_id
```

Comment:
Could probably use the principal_id attribute from the data resource here - unless there's a specific reason why it's a variable?